qiskit_experiments/base_experiment.py

```diff
@@ -201,59 +189,106 @@ def circuits(
             *N*-qubit experiment. The circuits mapped to physical qubits
             are obtained via the :meth:`transpiled_circuits` method.
         """
-        # NOTE: Subclasses should override this method with explicit
-        # kwargs for any circuit options rather than use `**circuit_options`.
-        # This allows these options to have default values, and be
-        # documented in the methods docstring for the API docs.
-
-    def transpiled_circuits(
-        self, backend: Optional[Backend] = None, **kwargs
-    ) -> List[QuantumCircuit]:
-        """Return a list of experiment circuits.
+        # NOTE: Subclasses should override this method using the `options`
+        # values for any explicit experiment options that effect circuit
+        # generation
+
+    @classmethod
+    def _default_experiment_options(cls) -> Options:
```

Comment:
Is there any special reason to keep these defaults as methods? I think we can also write them as class attributes. You assume some use case like this?
```python
options = super()._default_experiment_options
options['new_field'] = 1234
```

Reply:
This is how @mtreinish set it up so maybe he can comment more, but I think methods are easier to document for developers, and this is also guaranteed to return a fresh instance with each object so you never have to worry about copies and side effects which you would for a class attribute.

Reply:
Yeah, it's those reasons and that I also originally designed it to be part of the abstract backends interface. Doing it this way it was easier for it to be an `abstractmethod` and to have a defined attribute always present at init time in any class implementing the backend interface.

Reply:
Fair enough. Thanks.